src/Validation/URLValidationCron.php

```diff
@@ -96,15 +53,34 @@ public function __construct( BackgroundTaskDeactivator $background_task_deactiva
 	 * @param mixed[] ...$args Unused callback arguments.
 	 */
 	public function process( ...$args ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
-		$urls       = $this->scannable_url_provider->get_urls();
-		$sleep_time = $this->get_sleep_time();
+		$validation_queue_key = 'amp_url_validation_queue';
+		$validation_queue     = get_option( $validation_queue_key, [] );
 
-		foreach ( $urls as $url ) {
-			$this->url_validation_provider->get_url_validation( $url['url'], $url['type'] );
-			if ( $sleep_time ) {
-				sleep( $sleep_time );
+		if ( empty( $validation_queue ) || ! is_array( $validation_queue ) ) {
+			return [];
+		}
+
+		$limit = 5;
+		$count = 1;
+
+		foreach ( $validation_queue as $hash => $url ) {
+			if ( empty( $url['url'] ) || empty( $url['type'] ) ) {
+				continue;
+			}
+
+			$response = $this->url_validation_provider->get_url_validation( $url['url'], $url['type'] );
+			if ( empty( $response ) || is_wp_error( $response ) ) {
+				continue;
+			}
+
+			unset( $validation_queue[ $hash ] );
+			$count ++;
+			if ( $limit < $count ) {
+				break;
 			}
 		}
```

Comment:
Instead of looping over 5, I think this should rather just do one. If using the `in_array()` check suggested below, then this could be just:

```suggestion
		if ( empty( $validation_queue ) ) {
			return;
		}
		
		$entry = array_shift( $validation_queue );
		if ( isset( $entry['url'], $entry['type'] ) ) {
			$this->url_validation_provider->get_url_validation( $entry['url'], $entry['type'] );
		}
```